src/js_sync.hpp

```diff
@@ -192,24 +195,26 @@ class ClientResetAfterFunctor {
         return m_func;
     }
 
-    void operator()(SharedRealm before_realm, ThreadSafeReference after_realm_ref, bool)
+    void operator()(SharedRealm before_realm, ThreadSafeReference after_realm_ref, bool did_recover)
     {
-        HANDLESCOPE(m_ctx)
+        HANDLESCOPE(m_ctx);
 
         SharedRealm after_realm =
             Realm::get_shared_realm(std::move(after_realm_ref), util::Scheduler::make_default());
 
-
-        typename T::Value arguments[] = {
-            create_object<T, RealmClass<T>>(m_ctx, new SharedRealm(before_realm)),
-            create_object<T, RealmClass<T>>(m_ctx, new SharedRealm(after_realm)),
-        };
-        Function<T>::callback(m_ctx, m_func, 2, arguments);
+        typename T::Value arguments[3];
+        arguments[0] = create_object<T, RealmClass<T>>(m_ctx, new SharedRealm(before_realm));
+        arguments[1] = create_object<T, RealmClass<T>>(m_ctx, new SharedRealm(after_realm));
+        if (!m_ignore_recover) {
+            arguments[2] = Value<T>::from_string(m_ctx, did_recover ? "yes" : "no");
```

Comment:
I'd suggest this should simply be a `boolean`.

Reply:
An enum is specified in the technical design and will align better with other SDKs